subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -517,7 +518,25 @@ int nrf_cloud_rest_location_get(struct nrf_cloud_rest_context *const rest_ctx,
 	memset(&resp, 0, sizeof(resp));
 	init_rest_client_request(rest_ctx, &req, HTTP_POST);
 
-	req.url = request->disable_response ? API_GET_LOCATION_NO_REPLY : API_GET_LOCATION;
+	char url[sizeof(API_GET_LOCATION_CONFIG) + sizeof("false") * 3];
+	bool do_reply;
+
+	if (request->config) {
+		ret = snprintk(url, sizeof(url) - 1, API_GET_LOCATION_CONFIG,
+			       request->config->do_reply ? "true" : "false",
+			       request->config->fallback ? "true" : "false",
+			       request->config->hi_conf  ? "true" : "false");
+		if ((ret < 0) || (ret > (sizeof(url) - 1))) {
+			ret = -EIO;
+			goto clean_up;
+		}
+		do_reply = request->config->do_reply;
+	} else {
+		strncpy(url, API_GET_LOCATION, sizeof(url) - 1);
+		url[sizeof(url) - 1] = '\0';
+		do_reply = NRF_CLOUD_LOCATION_DOREPLY_DEFAULT;
```

Comment:
How about handling here cloud default values so the application doesn't need to think about whether to pass values (which might be cloud defaults) or NULL config?
It would resolve this also https://github.com/nrfconnect/sdk-nrf/pull/13417#discussion_r1441651810